src/main/java/org/jenkinsci/plugins/zap/ZAPDriver.java

```diff
@@ -1676,14 +1756,88 @@ private void setUpForcedUser(BuildListener listener, ClientApi clientApi, String
      * @throws ClientApiException
      * @throws UnsupportedEncodingException
      */
-    private String setUpAuthentication(BuildListener listener, ClientApi clientApi, String contextId, String loginURL, String username, String password, String loggedInIndicator, String extraPostData, String authMethod, String usernameParameter, String passwordParameter, String scriptName, ArrayList<ZAPAuthScriptParam> authScriptParams) throws ClientApiException, UnsupportedEncodingException {
-        if (authMethod.equals(FORM_BASED)) setUpFormBasedAuth(listener, clientApi, contextId, loginURL, loggedInIndicator, extraPostData, usernameParameter, passwordParameter);
-        else if (authMethod.equals(SCRIPT_BASED)) setUpScriptBasedAuth(listener, clientApi, authScriptParams, contextId, loginURL, loggedInIndicator, extraPostData, scriptName);
+    private String setUpAuthentication(BuildListener listener, ClientApi clientApi, String contextId, String loginURL, String username, String password, String loggedInIndicator, String loggedOutIndicator, String extraPostData, String authMethod, String usernameParameter, String passwordParameter, String scriptName, ArrayList<ZAPAuthScriptParam> authScriptParams) throws ClientApiException, UnsupportedEncodingException {
+        if (authMethod.equals(FORM_BASED)) setUpFormBasedAuth(listener, clientApi, contextId, loginURL, loggedInIndicator, loggedOutIndicator, extraPostData, usernameParameter, passwordParameter);
+        else if (authMethod.equals(SCRIPT_BASED)) setUpScriptBasedAuth(listener, clientApi, authScriptParams, contextId, loginURL, loggedInIndicator,  loggedOutIndicator, extraPostData, scriptName);
 
         return setUpUser(listener, clientApi, contextId, username, password);
     }
 
     /**
+     * parse xml file and add alert filter in the context.
+     *
+     * @param listener
+     *            of type BuildListener: the display log listener during the Jenkins job execution.
+     * @param clientApi
+     *            of type ClientApi: the ZAP client API to call method.
+     * @param xmlAlertsFilter
+     *            of type String: the name of the xml filter file.
+     * @throws ClientApiException
+     */
+    private void setUpContextAlertFilters(BuildListener listener,ClientApi clientApi,  String xmlAlertsFilter) throws ClientApiException, IOException, SAXException, ParserConfigurationException {
+        Utils.lineBreak(listener);
+        Utils.loggerMessage(listener, 0, "START : PARSING XML ALERT(s) FILTER File");
+
+        Path pathAuthScriptsDir = Paths.get(zapSettingsDir,NAME_ALERT_FILTERS_DIR_ZAP);
```

Comment:
Should be 
`Path pathAuthScriptsDir = Paths.get(evaluatedZapSettingsDir, NAME_ALERT_FILTERS_DIR_ZAP);`

Or else it will break when using environment variables.

Also the variable name `pathAuthScriptsDir` doesn't make sense (i assume this is from when you copy-pasted). Please change it to `pathAlertFiltersDir`

FINAL: `Path pathAlertFiltersDir = Paths.get(evaluatedZapSettingsDir, NAME_ALERT_FILTERS_DIR_ZAP);`